packages/solana/src/client.ts

```diff
@@ -0,0 +1,482 @@
+        }
+
+        if (solanaConfig.injected) {
+          if (!window?.originalSolana) {
+            return false
+          }
+        }
+
+        return false
+      },
+
+      disconnect: async () => {
+        const provider = SolStoreUtil.state.provider as Provider
+        const providerType = SolStoreUtil.state.providerType
+        localStorage.removeItem(SolConstantsUtil.WALLET_ID)
+        SolStoreUtil.reset()
+        if (providerType === ConstantsUtil.WALLET_CONNECT_CONNECTOR_ID) {
+          const WalletConnectProvider = provider
+          await (WalletConnectProvider as unknown as UniversalProvider).disconnect()
+        } else if (provider) {
+          provider.emit('disconnect')
+        }
+      },
+
+      signMessage: async (message: string) => {
+        const provider = SolStoreUtil.state.provider
+        if (!provider) {
+          throw new Error('connectionControllerClient:signMessage - provider is undefined')
+        }
+
+        const signature = await provider.request({
+          method: 'personal_sign',
+          params: [message, this.getAddress()]
+        })
+
+        return signature as string
+      }
+    }
+
+    super({
+      networkControllerClient,
+      connectionControllerClient,
+      defaultChain: SolHelpersUtil.getChainFromCaip(
+        chains,
+        typeof window === 'object' ? localStorage.getItem(SolConstantsUtil.CAIP_CHAIN_ID) : ''
+      ),
+      tokens: HelpersUtil.getCaipTokens(tokens),
+      _sdkVersion: _sdkVersion ?? `html-solana-${ConstantsUtil.VERSION}`,
+      ...w3mOptions
+    } as ScaffoldOptions)
+
+    this.chains = chains
+    this.syncRequestedNetworks(chains, chainImages)
+    SolStoreUtil.setProjectId(options.projectId)
+    const chain = SolHelpersUtil.getChainFromCaip(
+      chains,
+      typeof window === 'object' ? localStorage.getItem(SolConstantsUtil.CAIP_CHAIN_ID) : ''
+    )
+    if (chain) {
+      SolStoreUtil.setCurrentChain(chain)
+      SolStoreUtil.setCaipChainId(`${chain.name}:${chain.chainId}`)
+    }
+    this.syncNetwork(chainImages)
+
+    this.walletAdapters = createWalletAdapters()
+    this.WalletConnectConnector = new WalletConnectConnector({
+      relayerRegion: 'wss://relay.walletconnect.com',
+      metadata: {
+        description: 'Solana in Wallet Connect',
+        name: 'Wallet Connect',
+        icons: ['https://avatars.githubusercontent.com/u/37784886'],
+        url: 'https://web3modal.com'
+      },
+      autoconnect: false,
+      qrcode: true
+    })
+    SolStoreUtil.setConnection(
+      new Connection(chain?.rpcUrl ?? 'https://api.devnet.solana.com', 'recent')
+    )
+
+    SolStoreUtil.subscribeKey('address', () => {
+      this.syncAccount()
+    })
+
+    SolStoreUtil.subscribeKey('caipChainId', () => {
+      this.syncNetwork(chainImages)
+    })
+
+    if (typeof window === 'object') {
+      setTimeout(() => {
+        this.checkActiveProviders()
+        this.syncConnectors()
+      }, 500)
```

Comment:
any specific reason for this timeout? I'd add a comment on why if it's required.